core/node/libp2p/rcmgr_metrics.go

```diff
@@ -0,0 +1,239 @@
+package libp2p
+
+import (
+	"strconv"
+
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	rcmgr "github.com/libp2p/go-libp2p-resource-manager"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func createRcmgrMetrics() rcmgr.MetricsReporter {
+	const (
+		direction = "direction"
+		usesFD    = "usesFD"
+		protocol  = "protocol"
+		service   = "service"
+	)
+
+	connAllowed := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "libp2p_rcmgr_conns_allowed_total",
```

Comment:
do all these metrics need the "_total" suffix?

Reply:
I think it is a convention in case we ever add partials in the future
 http://127.0.0.1:5001/debug/metrics/prometheus has a lot of those:
 ```
$  curl -s http://127.0.0.1:5001/debug/metrics/prometheus | grep -c _total
361
 ```